test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -277,6 +277,28 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, machine *
 	dockerMachine.Status.Ready = true
 	conditions.MarkTrue(dockerMachine, infrav1.ContainerProvisionedCondition)
 
+	// set address in machine status
```

Comment:
This block (reconcile network) should go before L267 (reconcile ProviderID), otherwise, in case of errors network won't be reconciled due to the test at L172

Reply:
fixed